test/test_distributions.py

```diff
@@ -2239,6 +2251,15 @@ def test_binomial_shape(self):
         self.assertEqual(dist.log_prob(self.tensor_sample_1).size(), torch.Size((3, 2)))
         self.assertRaises(ValueError, dist.log_prob, self.tensor_sample_2)
 
+    def test_binomial_shape_vectorized_n(self):
```

Comment:
It would also be nice to draw a ton of samples and assert that `(sample <= total_count).all()`, just to make sure we got the masking correct. Maybe also numerically test the mean of `Binomial(total_count=torch.tensor([0,1,2,5,10]), probs=0.5)`?